File: backend/src/payments/shipping_config.py
"""Shipping policy: which countries get free shipping vs the flat fee.

Source of truth on the backend — mirrors `frontend/src/lib/shipping-config.ts`
and the country→region map in `frontend/src/lib/regional-pricing.ts`.
Keep these in sync.
"""

# Countries we explicitly ship to. Any country in this set gets FREE shipping.
# Any country NOT in this set (the fallback / WORLD region) pays FLAT_SHIPPING_FEE.
FREE_SHIPPING_COUNTRIES: frozenset[str] = frozenset({
    # UK
    "GB",
    # Germany
    "DE",
    # France & Spain
    "FR", "ES",
    # EU West
    "IT", "BE", "AT", "NL", "SE", "DK", "FI", "LU",
    # EU East
    "CZ", "PL", "GR", "PT", "IE", "RO", "BG", "HR",
    "SK", "SI", "HU", "EE", "LV", "LT",
    # USA
    "US",
    # Canada
    "CA",
    # Australia
    "AU",
})

# Flat shipping fee for non-free regions, charged in the Stripe session currency.
FLAT_SHIPPING_FEE: float = 8.0


def is_free_shipping_country(country_code: str | None) -> bool:
    if not country_code:
        return False
    return country_code.upper() in FREE_SHIPPING_COUNTRIES


def get_shipping_fee(country_code: str | None) -> float:
    return 0.0 if is_free_shipping_country(country_code) else FLAT_SHIPPING_FEE

File: frontend/src/lib/shipping-config.ts
// Shipping policy: which regions get free shipping vs the flat fee.
//
// Source of truth — edit this file to change shipping behavior.
// Mirrored on the backend in `backend/src/payments/shipping_config.py`;
// keep the two in sync.

import type { Region } from "./regional-pricing";

// Regions in this list ALWAYS get free shipping. Any region NOT in this
// list (today: only WORLD) pays the flat fee in their local currency.
export const FREE_SHIPPING_REGIONS: readonly Region[] = [
  "UK",
  "DE",
  "FR_ES",
  "EU_WEST",
  "EU_EAST",
  "US",
  "CA",
  "AU",
];

// Flat shipping fee for non-free regions, in each region's local currency.
export const FLAT_SHIPPING_FEE = 8;

export function isFreeShippingRegion(region: Region): boolean {
  return FREE_SHIPPING_REGIONS.includes(region);
}

export function getShippingFee(region: Region): number {
  return isFreeShippingRegion(region) ? 0 : FLAT_SHIPPING_FEE;
}
